src/PerfView/PerfViewData.cs

```diff
@@ -3402,6 +3402,454 @@ private void CreateCSVFile(List<AspNetRequest> requests, string csvFileName)
+                            break;
+                        }
+
+                        if (incompleteRequests.TryGetValue(ANCHostingRequest.GetIndexingKeyFromEvent(traceEvent), out request))
+                        {
+                            // request exists, update its properties
+                            request.StartTimeRelativeMSec = traceEvent.TimeStampRelativeMSec;
+                            request.HasStart = true;
+                            request.Method = (string)traceEvent.PayloadByName("method");
+                            request.Path = (string)traceEvent.PayloadByName("path");
+                            if (String.IsNullOrEmpty(request.Path))
+                                request.Path = "/";
+                        }
+                        else
+                        {
+                            // request does not exist yet, create and track it
+                            request = new ANCHostingRequest(traceEvent);
+                            request.Method = (string)traceEvent.PayloadByName("method");
+                            request.Path = (string)traceEvent.PayloadByName("path");
+                            if (String.IsNullOrEmpty(request.Path))
+                                request.Path = "/";
+                            request.StartTimeRelativeMSec = traceEvent.TimeStampRelativeMSec;
+                            request.HasStart = true;
+                            incompleteRequests.Add(request.IndexingKey, request);
+                        }
+
+                        break;
+
+                    case _eventRequestStop:
+                        if (traceEvent.ActivityID.Equals(Guid.Empty))
+                        {
+                            eventsWithNoActivityIdCount++;
+                            break;
+                        }
+
+                        if (incompleteRequests.TryGetValue(ANCHostingRequest.GetIndexingKeyFromEvent(traceEvent), out request))
+                        {
+                            // request exists, update its properties
+                            request.EndTimeRelativeMSec = traceEvent.TimeStampRelativeMSec;
+                            request.HasStop = true;
+
+                            // if the request is complete then move it to the completeRequests List
+                            if (request.IsComplete)
+                            {
+                                incompleteRequests.Remove(request.IndexingKey);
+                                completeRequests.Add(request);
+                            }
+                        }
+                        else
+                        {
+                            // request does not exist yet, create and track it
+                            request = new ANCHostingRequest(traceEvent);
+                            request.EndTimeRelativeMSec = traceEvent.TimeStampRelativeMSec;
+                            request.HasStop = true;
```

Comment:
Just thinking about this a little further - I think that we want this else block to not add to incompleteRequests.  If you want to try and track requests that have a stop but no start, that seems OK, but this code would allow a start to show up after a stop and I still think we don't want that.